### PR TITLE
`kubectl label` should support resource builder

### PR DESCRIPTION
**Pull request by smarterclayton**
Allow applying labels to all resources, by existing selector, and
soon allow multiple selection.

### CODE REVIEW

**Comment by smarterclayton** [comment]
Flaked on TestSyncPods, not a real failure.

File: hack/test-cmd.sh [hunk 1]
```diff
@@ -307,11 +307,11 @@ for version in "${kube_api_versions[@]}"; do
   # Post-condition: name is still valid-pod
   kube::test::get_object_assert 'pod valid-pod' "{{.${labels_field}.name}}" 'valid-pod'
 
-  ### --overwrite must be used to overwrite existing label
+  ### --overwrite must be used to overwrite existing label, can be applied to all resources
   # Pre-condition: name is valid-pod
   kube::test::get_object_assert 'pod valid-pod' "{{.${labels_field}.name}}" 'valid-pod'
   # Command
-  kubectl label --overwrite pods valid-pod name=valid-pod-super-sayan "${kube_flags[@]}"
+  kubectl label --overwrite pods --all name=valid-pod-super-sayan "${kube_flags[@]}"
```
Comment (jlowdermilk) [inline hack/test-cmd.sh]: Nit: s/sayan/saiyan

Comment (smarterclayton) [inline hack/test-cmd.sh]: Does that mean something?  :)

----- Original Message -----

File: pkg/kubectl/cmd/label.go [hunk 2]
```diff
@@ -152,52 +150,80 @@ func labelFunc(obj runtime.Object, overwrite bool, resourceVersion string, label
...
+	one := false
+	r := b.Do().IntoSingular(&one)
+	if err := r.Err(); err != nil {
 		return err
 	}
-	overwrite := util.GetFlagBool(cmd, "overwrite")
-	resourceVersion := util.GetFlagString(cmd, "resource-version")
+	// only apply resource version locking on a single resource
+	if !one {
+		resourceVersion = ""
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/label.go]: I think it's clearer to exit with an error if the user specified both resource version and multiple resources. Silently ignoring the resource version might confuse.

Comment (smarterclayton) [inline pkg/kubectl/cmd/label.go]: True.

File: hack/test-cmd.sh (hunk 1 above)
Comment (jlowdermilk) [inline hack/test-cmd.sh]: Entirely frivolous nit. I figured you were making a dbz reference :P

Comment (smarterclayton) [inline hack/test-cmd.sh]: I was just copying the previous person's incorrectly spelled dbz reference.

----- Original Message -----

File: pkg/kubectl/cmd/label.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/kubectl/cmd/label.go]: Fixed.

----- Original Message -----

File: hack/test-cmd.sh (hunk 1 above)
Comment (jlowdermilk) [inline hack/test-cmd.sh]: We can leave it for a more enthusiastic and less busy dbz fan to correct.

**Comment by jlowdermilk** [comment]
LGTM. kicking travis

**Comment by smarterclayton** [comment]
it's green

**Pull request merged**